src/granite_orm/validators.cr

```diff
@@ -25,7 +36,25 @@ module Granite::ORM::Validators
     end
   end
 
+  # Analyze validation blocks and procs
+  # Note: This method checks two type of validation macros
+  #
+  # By example:
+  # ```
+  # validate :name, "name can't be blank" do
+  #   !name.to_s.blank?
+  # end
+  #
+  # validate :name, "name can't be blank", -> (user : User) do
+  #   !user.name.to_s.blank?
+  # end
+  # ```
```

Comment:
It's probably a good idea to have this comment which shows the validation syntax, but I don't think it belongs on the `valid?` method. Perhaps on the module?